fix Java doc structure and add default configurations (#634)
Amplitude-Developer-Docs/amplitude-dev-center: docs/data/sdks/java/index.md

client.init("YOUR_API_KEY");
```

### Configuration

???config "Configuration Options"
    | <div class="big-column">Name</div>  | Description | Default Value |
    | --- | --- | --- |

>     | `serverUrl` | The server url events upload to. | `https://api2.amplitude.com/2/httpapi` |

Reviewer: ```suggestion
    | `serverUrl` | The server url events are uploaded to. | `https://api2.amplitude.com/2/httpapi` |
```

>     | `minIdLength` | The minimum length for user id or device id. | `5` |

Reviewer 2: is this supported in Java now?

Author: Yes, they setup way is a little bit different. 
```
    amplitude.setOptions(new Options()
            .addHeader("Custom Header", "value")
            .setMinIdLength(5));
```
I will update it to options.minIdLength.

Author: [Pr](https://github.com/amplitude/Amplitude-Java/pull/63) for reference.

    | `isBatchMode` |  Whether to use batch api. | `false` |

>     | `logMode` |  Whether to use batch api. | `LogMode.ERROR` |

Reviewer: Typo?

client.init("YOUR_API_KEY");
```

### Configuration

???config "Configuration Options"
    | <div class="big-column">Name</div>  | Description | Default Value |
    | --- | --- | --- |
    | `serverUrl` | The server url events are uploaded to. | `https://api2.amplitude.com/2/httpapi` |
    | `isBatchMode` |  Whether to use batch api. | `false` |
    | `logMode` |  The level at which to filter out debug messages. | `LogMode.ERROR` |
    | `eventUploadThreshold` |  SDK will attempt to upload once unsent event count exceeds the event upload threshold or reach eventUploadPeriodSeconds interval. | `10` |
    | `eventUploadPeriodMillis` | The amount of time SDK will attempt to upload the unsent events to the server or reach eventUploadThreshold threshold. | `10 seconds` |
    | `callbacks` | Event callback which are triggered after event sent. | `null`|
    | `proxy` | custom proxy for https requests. | `Proxy.NO_PROXY`|
    | `flushTimeout` | events flushing thread timeout in milliseconds. | `0` |

>     | `options.minIdLength` | The minimum length for user id or device id. | `5` |

Reviewer 3: Trying to decide the best way to label this. If the customer should use a setter `Options.setMinIdLength()` should we show the method name instead? I saw that `options.minIdLength` is also public so the current `options.minIdLength` could still make sense. wdyt @yuhao900914 @qingzhuozhen?

I had a similar case in Legacy iOS with `setIdentifyUploadPeriodSeconds`. We should decide on how to make this uniform for all SDKs.
https://www.docs.developers.amplitude.com/data/sdks/ios/#configuration

Reviewer 2: For the next-gen SDKs this is part of the configuration. For the legacy ones, normally it is setters, and Java SDK is one we introduced the options hence we see the difference here.

>     | `options.headers` | The custom headers. | `{"Content-Type", "application/json",

Reviewer 2: The format show in preview is a bit broken, can you fix this? https://pr-634.d19s7xzcva2mw3.amplifyapp.com/data/sdks/java/#configuration

client.init("YOUR_API_KEY");
```

### Configuration

???config "Configuration Options"
    | <div class="big-column">Name</div>  | Description | Default Value |
    | --- | --- | --- |
    | `setServerUrl` | The server url events are uploaded to. | `https://api2.amplitude.com/2/httpapi` |

>     | `useBatchMode` |  Whether to use [batch api](../../../analytics/apis/batch-event-upload-api/#batch-event-upload). By default, the SDK will use the default `serverUrl`. | `false` |

Reviewer 4: ```suggestion
    | `useBatchMode` |  Whether to use [batch API](../../../analytics/apis/batch-event-upload-api/#batch-event-upload). By default, the SDK will use the default `serverUrl`. | `false` |
```

    | `setLogMode` |  The level at which to filter out debug messages. | `LogMode.ERROR` |
    | `setEventUploadThreshold` |  SDK will attempt to upload once unsent event count exceeds the event upload threshold or reach eventUploadPeriodSeconds interval. | `10` |
    | `setEventUploadPeriodMillis` | The amount of time SDK will attempt to upload the unsent events to the server or reach eventUploadThreshold threshold. | `10 seconds` |
    | `setCallbacks` | Event callback which are triggered after event sent. | `null`|
    | `setProxy` | Custom proxy for https requests. | `Proxy.NO_PROXY`|
    | `setFlushTimeout` | Events flushing thread timeout in milliseconds. | `0` |
    | `setOptions` | A dictionary of key-value pairs that represent additional instructions for server save operation.| Please check [available options](/#options). |

### Options

???config "Availabe Options"
    | <div class="big-column">Name</div>  | Description | Default Value |
    | --- | --- | --- |
    | `Options.setMinIdLength` | Set the minimum length for user id or device id. | 5 |

>     | `Options.setHeaders` | Set the custom headers. | `{"Content-Type", "application/json","Accept", "application/json"}` |

Reviewer: Minor, missing space `json","Accept`, also to confirm, should this be Map or JSON? (https://github.com/amplitude/Amplitude-Java/blob/0219df6aa61e1414326fe759ab1dbd569daa83c7/src/main/java/com/amplitude/Options.java#L23)

    | `Options.addHeaders` | Add more custom headers. | `{"Content-Type", "application/json", "Accept", "application/json"}` |

#### Configure batching behavior

> To support high performance environments, the SDK sends events in batches. Every event logged by `logEvent` method is queued in memory. Events are flushed in batch in background. You can customize batch behavior with `setEventUpdloadThreshfold` and `setEventUploadPeriodMillis`. By default, the serverUrl will be `https://api2.amplitude.com/2/httpapi`. For customers who want to send large batches of data at a time. please check the `useBatchMode`. Both the regular mode and the batch mode use the same events upload threshold and flush time intervals.

Reviewer: Minor, `time. please check` -> `time, please check`

> To support high performance environments, the SDK sends events in batches. Every event logged by `logEvent` method is queued in memory. Events are flushed in batch in background. You can customize batch behavior with `setEventUploadThreshfold` and `setEventUploadPeriodMillis`. By default, the serverUrl will be `https://api2.amplitude.com/2/httpapi`. For customers who want to send large batches of data at a time, please check the `useBatchMode`. Both the regular mode and the batch mode use the same events upload threshold and flush time intervals.

Reviewer 4: typo: `setEventUploadThreshfold ` to `setEventUploadThreshold`
```suggestion
To support high performance environments, the SDK sends events in batches. Every event logged by `logEvent` method is queued in memory. Events are flushed in batch in background. You can customize batch behavior with `setEventUploadThreshold` and `setEventUploadPeriodMillis`. By default, the serverUrl will be `https://api2.amplitude.com/2/httpapi`. For customers who want to send large batches of data at a time, please check the `useBatchMode`. Both the regular mode and the batch mode use the same events upload threshold and flush time intervals.
```

Author: A hard one! Thanks.

> To support high performance environments, the SDK sends events in batches. Every event logged by `logEvent` method is queued in memory. Events are flushed in batch in background. You can customize batch behavior with `setEventUpdloadThreshfold` and `setEventUploadPeriodMillis`. By default, the serverUrl will be `https://api2.amplitude.com/2/httpapi`. For customers who want to send large batches of data at a time. please check the `useBatchMode`. Both the regular mode and the batch mode use the same events upload threshold and flush time intervals.

Author: Added more words for avoiding confusion between `batch` and `batch mode`. But it's has some overlap with the following doc. Let me know what do you think.
<img width="783" alt="Screen Shot 2023-03-15 at 11 53 33 AM" src="https://user-images.githubusercontent.com/13028329/225414782-905059f8-8ffc-4147-a3c8-85082c08f4c4.png">

Reviewer 4: How about explicitly saying the main difference between two modes is which `serverUrl` they use? 

> For customers who want to send large batches of data at a time, set`useBatchMode` to `true` to set `setServerUrl` to batch event upload api `https://api2.amplitude.com/batch`.

Reviewer 3: Thanks @Mercy811 I think mentioning the separate endpoint is a good idea
